zerver/lib/events.py

```diff
@@ -193,7 +193,8 @@ def fetch_initial_state_data(
             state["drafts"] = user_draft_dicts
 
     if want("muted_topics"):
-        state["muted_topics"] = [] if user_profile is None else get_topic_mutes(user_profile)
+        if event_types is None or not want("user_topic"):
+            state["muted_topics"] = [] if user_profile is None else get_topic_mutes(user_profile)
 
```

Comment:
If a client sends in `event_types = None` (meaning, it is requesting for all the `page_params`), then is the intention to send the `muted_topics` events, or not? Because this could be an old client unknowing of `user_topic` event type, we should send it, right?

Reply:
@abhijeetbodas2001 Yes, the intention here is to send both `muted_topics` and `user_topic` when a client sends in `event_types = None`.

Reply:
OK, thanks! So is the following table correct? (where in the first 3 cases, `event_types != None`)
|Client request|What to send|
|-|-|
|Specifically `user_topic`|*Only* `user_topic`|
|Both|*Only* `user_topic`|
|Specifically `muted_users`|*Only* `muted_users`|
|None (ie the default `page_params`, with `event_types=None`|Both|

Apologies if this is something that has already been discussed -- I haven't followed the discussion on this closely, but I found the `if` blocks here to be a bit difficult to read/understand (especially since the `want` function itself also checks for `event_types=None`) and so am wondering if some simpler implementation is possible.

Reply:
In case, when a client requests for both `user_topic` and `muted_topics`, we send only `user_topic` events. In that case, both `want("user_topic")` and `want("muted_topics")` evaluate to true. So, to decide whether to send the `muted_topics` state, we have to check whether it has been requested alongside all the page_params, or just `user_topic`. So, we check if `event_types==None`. If that's true, we send `muted_topics`. If that's false, we send `muted_topics`, if the user doesn't `want("user_topic")`.

Reply:
I think that's potentially right, but can you resummarize this in "#api design" so we can get confirmation from the mobile developers that this interface makes sense to them?

We may end up coming up with a clever way to implement this logic with `want()`.